feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java

```diff
@@ -138,18 +140,15 @@ public void start(Stage stage) {
         featureLayer.setDefinitionExpression("");
       });
 
-      // check that the feature layer has loaded, then add a draw status changed listener to the mapview to count 
+      // check that the feature layer has loaded, then add a listener to the mapview to count
       // the features visible in the viewpoint extent every time the map is redrawn
       featureLayer.addDoneLoadingListener(() -> {
         if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
-          mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
-            // if the draw status is in progress, show the progress indicator and set the count label text
-            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS) {
-              progressIndicator.setVisible(true);
+          mapView.drawStatusProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue == DrawStatus.IN_PROGRESS) {
+              // set the count label text
```

Comment:
Is there anything we can do with this feature count label - it is being set to "loading" at the start and then "updating" when the mapview is drawing. It may not like it though if you then set the text on line 166, so not to worry if it doesn't work but could be worth a try if you haven't already.

Reply:
It seems bound values cannot be set. There could be a workaround to change the binding when a condition is met and having two different bindings for the label text.